added endpoint to updated team's points

mchlcui opened the pull request:
👷 **Changes:**
I added an endpoint for updating the teams' points field through increments and decrements. 

💭 **Notes:**
It would potentially be beneficial to have an endpoint to set the points field, putting it on hold for the time sake of DataVerse. 

<img width="1045" alt="image" src="https://github.com/user-attachments/assets/c1992b83-c3c7-4085-8528-e827c2c4b2f1">

**Wait! Before you merge, have you checked the following:**
- [ ] PR is has approving review(s)

AllanT102 commented on services/teams/handler.js:
```diff
@@ -21,6 +21,52 @@ import db from "../../lib/db.js";
+      data.year
+    );
+
+    console.log(team);
```
remove log

AllanT102 commented:
how are we handling multiple submissions? ie. if someoen tries to resubmit then it shouldn't increment points

mchlcui commented:
> how are we handling multiple submissions? ie. if someoen tries to resubmit then it shouldn't increment points

This is in progress, I will be adding a field in the TEAMS table that includes a set of the questions that has been answered, so a team cannot be rewarded extra points

AllanT102 commented:
> > how are we handling multiple submissions? ie. if someoen tries to resubmit then it shouldn't increment points
> 
> This is in progress, I will be adding a field in the TEAMS table that includes a set of the questions that has been answered, so a team cannot be rewarded extra points

Instead of introducing a new field in the teams table, are we able to reuse the scannedQrs field or qrscsanhistory field in the registrations table? I forget what the field is called. 

If that's too much work then just create new field then

mchlcui commented:
fixed, now have endpoints that uses the scannedQrs field to store correctly answered questions. also added an endpoint so questions can be added at once during a quiz submission instead of making an API call for each question answered correctly.

AllanT102 commented on services/teams/helpers.js:
```diff
@@ -217,6 +217,54 @@ export default {
+        }
+
+        if (totalPoints < 0) {
+          team.pointsSpent += Math.abs(totalPoints);
```
what is pointSpent?

mchlcui commented on services/teams/serverless.yml:
```diff
@@ -55,6 +55,27 @@ provider:
+          method: put
+          cors: true
+  addQRScan:
+    handler: handler.addQRScan
```
no thats a function in the handler, a deprecated call. I removed it though since i wont be using it
